### PR TITLE
Capture more dumps on failure.

### PR DESCRIPTION
**Pull request by gianm**
Capture `*.hprof`, `hs_err_pid*`, `replay_pid*`, and `core.NNNN` on failure.

### CODE REVIEW

**Review by pranavbhole — APPROVED** [review]

File: .github/workflows/reusable-unit-tests.yml [hunk 1]
```diff
@@ -117,23 +117,26 @@ jobs:
...
 
-      - name: Collect tarball hprof dumps if they exist on failure
-        if: ${{ failure() && env.found_hprof == 'true' }}
+      - name: Collect dumps if they exist on failure
+        if: ${{ failure() && env.found_dumps == 'true' }}
         run: |
-          tar cvzf ${RUNNER_TEMP}/hprof-dumps.tgz ${GITHUB_WORKSPACE}/target/*.hprof
+          find "${GITHUB_WORKSPACE}" \
+            \( -name '*.hprof' -or -name 'hs_err_pid*' -or -name 'replay_pid*' -or -regex '.*/core\.[0-9]*' \) \
+            -exec tar -cvzf ${RUNNER_TEMP}/hprof-dumps.tgz {} +
```
Comment (abhishekrb19) [inline .github/workflows/reusable-unit-tests.yml:136]: Perhaps just dumps or failure-dumps:
```suggestion
            -exec tar -cvzf ${RUNNER_TEMP}/dumps.tgz {} +
```

File: .github/workflows/reusable-unit-tests.yml [hunk 2]
```diff
@@ -117,23 +117,26 @@ jobs:
...
 
       - name: Upload hprof dumps to GitHub if they exist on failure
```
Comment (abhishekrb19) [inline .github/workflows/reusable-unit-tests.yml:138]:
```suggestion
      - name: Upload dumps to GitHub if they exist on failure
```

File: .github/workflows/reusable-unit-tests.yml [hunk 3]
```diff
@@ -117,23 +117,26 @@ jobs:
...
-        if: ${{ failure() && env.found_hprof == 'true' }}
+        if: ${{ failure() && env.found_dumps == 'true' }}
         uses: actions/upload-artifact@master
         with:
           name: Hprof-${{ inputs.group }} hprof dumps (Compile=jdk${{ inputs.build_jdk }}, Run=jdk${{ inputs.runtime_jdk }})
```
Comment (abhishekrb19) [inline .github/workflows/reusable-unit-tests.yml:142]:
```suggestion
          name: ${{ inputs.group }} dumps (Compile=jdk${{ inputs.build_jdk }}, Run=jdk${{ inputs.runtime_jdk }})
```

the path below to `{{ runner.temp }}/dumps.tgz` to match the above.

File: .github/workflows/reusable-unit-tests.yml (hunk 1 above)
Comment (gianm) [reply .github/workflows/reusable-unit-tests.yml]: Renamed to `failure-dumps.tar.gz`

File: .github/workflows/reusable-unit-tests.yml (hunk 2 above)
Comment (gianm) [reply .github/workflows/reusable-unit-tests.yml]: Renamed.

File: .github/workflows/reusable-unit-tests.yml (hunk 3 above)
Comment (gianm) [reply .github/workflows/reusable-unit-tests.yml]: Renamed to `failure-dumps.tar.gz`

**Review by abhishekrb19 — APPROVED** [review]
Thanks!

**Pull request merged**